experimental/ComplexReflectionGroups/src/ComplexReflection.jl

```diff
@@ -0,0 +1,344 @@
+        return false, nothing
+    end
+
+    # zeta needs to be of finite order (so that w is of finite order), i.e. zeta needs
+    # to be a root of unity
+    b, d = is_root_of_unity_with_data(zeta)
+
+    if b == false
+        return false, nothing
+    end
+
+    # Now, determine the coroot.
+    # We have h*(I-w) = 0 for any h \in H. This means that h*v = 0 for any column v of I-w.
+    # We thus take L_H to be a non-zero column of I-w, considered as a row vector.
+    # Normalization yields the coroot of w with respect to alpha.
+    i=1
+    while is_zero(A[:,i])
+        i += 1
+    end
+    L_H = V(A[:,i])
+    alpha_check = (1-zeta)//canonical_pairing(L_H,alpha) * L_H
+
+    w_data = ComplexReflection(K, V, w, alpha, R, Rincl, alpha_check, linear_form(alpha_check), H, Hincl, [ Hincl(h) for h in gens(H) ], zeta, d, is_unitary(w))
+
+    return true, w_data
+
+end
+
+function is_complex_reflection(w::MatElem{T}) where T <: QQAlgFieldElem
+
+    b,data = is_complex_reflection_with_data(w)
+    return b
+
+end
+
+function complex_reflection(w::MatElem{T}) where T <: QQAlgFieldElem
+
+    b,w_data = is_complex_reflection_with_data(w)
+
+    if !b
+        throw(ArgumentError("Matrix is not a reflection"))
+    end
+
+    return w_data
+
+end
+
+###########################################################################################
+# Construction of a complex reflection from matrix group element
+###########################################################################################
+function is_complex_reflection_with_data(w::MatrixGroupElem{T}) where T <: QQAlgFieldElem
+
+    G = parent(w)
+
+    # If G has reflections assigned (see function below), try to find w in the list and 
+    # return the data.
+    if has_attribute(G, :complex_reflections)
+        refls = collect(get_attribute(G, :complex_reflections))
+        i = findfirst(g->matrix(g)==matrix(w), refls)
+        if i != nothing
+            w_data = refls[i]
+            return true, w_data
+        end
+    end
+
+    # Otherwise compute
+    return is_complex_reflection_with_data(matrix(w))
+end
+
+function is_complex_reflection(w::MatrixGroupElem{T}) where T <: QQAlgFieldElem
+    b, w_data = is_complex_reflection_with_data(w)
+    return b
+end
+
+function complex_reflection(w::MatrixGroupElem{T}) where T <: QQAlgFieldElem
+
+    b,w_data = is_complex_reflection_with_data(w)
+
+    if !b
+        throw(ArgumentError("Group element is not a reflection"))
+    end
+
+    return w_data
+
+end
+
+###########################################################################################
+# Determining all reflections of a matrix group.
+###########################################################################################
+function complex_reflections(G::MatrixGroup{T}) where T <: QQAlgFieldElem
+   
+    if has_attribute(G, :complex_reflections)
+        return get_attribute(G, :complex_reflections)
+    end
+
+    refls = Set(ComplexReflection[])
```

Comment:
I was not sure really. I later want to define maps on the set of reflections. Can I do this if this is a vector? But yeah, for easy access vectors are better. Opinions?

Reply:
In this use-case of maps, vectors are even better, since you have a first map, second map etc. (even though it's only internal). So your maps may just be permutations

Reply:
Will turn into vector.